dali/kernels/imgproc/convolution/cutlass/threadblock/predicated_tile_iterator.h

```diff
@@ -299,6 +438,49 @@ class PredicatedTileIterator<Shape_, Element_, layout::PitchLinear, AdvanceRank,
     load_with_byte_offset(frag, pointer_offset * sizeof_bits<Element>::value / 8);
   }
 
+  CUTLASS_DEVICE
+  void load_access_elements(AccessType *frag_ptr, int idx) {
+    // This calculates the logical coordinate of the beggining of access
+    TensorCoord current_coord = address_iterator_.get_current_coord();
+
+    // Distance from diagonal in the dim in which we place window (inner - column, outer - row)
+    int dist_diag = 0;
+    // Distance from first row/column for inner/outer convolution
+    int dist_first = 0;
+    // Distance from last row/column for inner/outer convolution
+    int dist_last = 0;
+    int row = current_coord.strided();
+    int col = current_coord.contiguous();
+    int height = address_iterator_.get_extent().strided();
+    int width = address_iterator_.get_extent().contiguous();
+    if (kIsInnerConv) {
+      // we generate matrix by placing windows vertically, so the row is the major coordinate
+      dist_diag = row - col;  // negative above coordinate
+      dist_first = row;
+      dist_last = height - 1 - row;
+    } else {
+      // we generate matrix by placing windows horizontally and there is no channel-based spacing
+      dist_diag = col - row;
+      dist_first = col;
+      dist_last = width - 1 - col;
+    }
+    // this is the starting element of the window, for inner-conv vector load goes in negative order
+    int window_element = dist_diag;
+    int radius = (window_size_ / 2) * Channels();
+
+    bool is_used = any_valid<kIsInnerConv>(window_element, radius) && address_iterator_.valid();
```

Comment:
any_valid is confusing here. rename to is_within_radius, or is_in_window or something like that

Reply:
done